packages/checkbox-group/src/vaadin-checkbox-group.js

```diff
@@ -305,8 +318,9 @@ class CheckboxGroup extends FieldMixin(FocusMixin(DisabledMixin(ElementMixin(The
    * @private
    */
   __onCheckboxCheckedChanged(event) {
-    const checkbox = event.target;
+    this.dirty = true;
```

Comment:
This marks the group as `dirty` when its `value` is changed programmatically.

Reply:
Good catch, thanks! I guess it makes sense to add a corresponding test to each component.

Reply:
Fixed and added tests.